### PR TITLE
fix all warnings about usage of undeclared functions

### PR DESCRIPTION
**Pull request by DerDakon**

### CODE REVIEW

File: seek.h [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
+
+
+static inline int seek_end(int fd)
+{
+  if (lseek(fd, 0, SEEK_END) == -1)
+    return -1;
+  return 0;
+}
+
+static inline int seek_trunc(int fd, seek_pos pos)
```
Comment (schmonz) [inline seek.h:29]: Out of curiosity, why not `#define seek_trunc(fd, pos) (ftruncate(fd, pos))`, at least for this one with no branching logic?

File: maildir2mbox.c [hunk 2]
```diff
@@ -14,6 +15,8 @@
 #include "myctime.h"
 #include "maildir.h"
 
+extern int rename(const char *, const char *);
```
Comment (schmonz) [inline maildir2mbox.c:18]: Are you already plotting the demise of `puts()`, so we can come back for `<stdio.h>` on these?

Comment (DerDakon) [reply maildir2mbox.c:18]: I would love to, but I think ```puts()``` is one of the things that is likely actually used by external patches.

File: seek.h (hunk 1 above)
Comment (DerDakon) [reply seek.h:29]: sure, would be consistent with other changes.

File: maildir2mbox.c (hunk 2 above)
Comment (schmonz) [reply maildir2mbox.c:18]: Yes, that’s true.

I’m not worried we’ll forget to come back to this someday (when external patches are way less common). Pretty soon, when we have hardly any `extern` left, this’ll stick out. :-)

**Pull request merged**